third-party/gmp/Makefile

```diff
@@ -11,11 +11,24 @@ include $(CHPL_MAKE_HOME)/make/Makefile.base
 GMP_UNPACKED_DIR = gmp-$(GMP_VERSION)
 GMP_TARBALL = $(GMP_UNPACKED_DIR)a.tar.bz2
 
+#
+# Cray X* builds are cross-compilations, as are Intel Phi KNC builds.
+#
+GMP_CROSS_COMPILED=no
+ifneq (, $(filter cray-x%,$(CHPL_MAKE_TARGET_PLATFORM)))
+CHPL_GMP_CFG_OPTIONS += --build=x86_64-unknown-linux-gnu \
+                        --host=x86_64-cnl-linux-gnu
+GMP_CROSS_COMPILED=yes
+else ifeq ($(CHPL_MAKE_TARGET_ARCH),knc)
+CHPL_GMP_CFG_OPTIONS += --host=x86_64-unknown-linux-gnu
```

Comment:
Another possibility would be to change the '*-cle-*' ones to only throw --host and not --build. Then at least everything would look the same. There's precedent for doing things this way, now that I look. NERSC's web page on babbage/cori says that this is one way to deal with the cross-compile issue for knl Cray systems when using autoconf. I just tried this (leaving off the --build option) with the hwloc build and it worked properly.